src/components/TimeseriesChart/CursorOverview.tsx

```diff
@@ -0,0 +1,178 @@
+import moment from 'moment';
+import numeral from 'numeral';
+import React from 'react';
+import styled from 'styled-components';
+import { LegendValue, LegendValues, RulerConfig } from './TimeseriesChart';
+
+const Container = styled.div`
+  position: fixed;
+  top: 0;
+  left: 0;
+  z-index: 100;
+`;
+
+const Overview = styled.div`
+  position: absolute;
+  top: 0;
+  margin-left: -180px;
+  width: 140px;
+  background: #fff;
+  padding: 8px 16px 10px;
+  pointer-events: none;
+  opacity: 0.9;
+  border-radius: 4px;
+  border: 1px solid #888888;
+  box-sizing: border-box;
+`;
+
+const DateContainer = styled.div`
+  position: absolute;
+  top: 0;
+  margin-left: -260px;
+  width: 220px;
+  color: black;
+  background: #fff;
+  padding: 8px 16px 10px;
+  pointer-events: none;
+  opacity: 0.9;
+  border-radius: 4px;
+  border: 1px solid #888888;
+  box-sizing: border-box;
+`;
+
+const Tag = styled.div`
+  color: black;
+  &:before {
+    content: '';
+    display: inline-block;
+    width: 10px;
+    height: 10px;
+    border-radius: 100%;
+    margin: 0 8px 0 2px;
+    background: ${props => props.color};
+  }
+`;
+
+const formattedDate = (timestamp: number) =>
+  moment(timestamp).format('MMM D, YYYY HH:mm:ss');
+
+export interface CursorOverviewStyles {
+  container?: React.CSSProperties;
+}
+interface CursorOverviewState {
+  overviewContainer: HTMLElement | null;
+}
+export interface CursorOverviewProps {
+  styles?: CursorOverviewStyles;
```

Comment:
we're trying to keep required props before an optional inside interface

Reply:
Done